jbehave-support-core/src/main/java/org/jbehavesupport/core/report/extension/ServerLogXmlReporterExtension.java

```diff
@@ -42,75 +76,70 @@ public Long getPriority() {
 
     @Override
     public void print(final Writer writer, final ReportContext reportContext) {
-        Map<String, List<SshTemplate>> sshTemplates = getSshTemplates();
-        for (String qualifier : sshTemplates.keySet()) {
-            printBegin(writer, SYSTEM, getSshQualifierAttributes(qualifier));
-            for (SshTemplate sshTemplate : sshTemplates.get(qualifier)) {
-                printBegin(writer, LOG, getSshTemplateAttributes(reportContext, sshTemplate));
-                try {
-                    SshLog sshLog = sshTemplate.copyLog(reportContext.startExecutionZoned(), reportContext.endExecutionZoned());
-                    printCData(writer, sshLog.getLogContents());
-                } catch (Exception e) {
-                    printBegin(writer, FAIL);
-                    printCData(writer, ExceptionUtils.getStackTrace(e));
-                    printEnd(writer, FAIL);
-                }
+        printContent(writer);
+        logContents.clear();
+    }
+
+    private void printContent(final Writer writer) {
+        logContents.entrySet().stream().forEach(
+            entry -> {
+                final MultiKey key = entry.getKey();
+                printBegin(writer, SYSTEM, getSshQualifierAttributes(key.getKey(0).toString()));
+                printBegin(writer, LOG, getSshAttributesFromKey(key));
+                printLogContent(writer, key.getKey(0).toString(), entry.getValue());
                 printEnd(writer, LOG);
+                printEnd(writer, SYSTEM);
             }
-            printEnd(writer, SYSTEM);
-        }
+        );
     }
 
-    private Map<String, String> getSshQualifierAttributes(String qualifier) {
-        Map<String, String> sshQualifierAttributes = new HashMap<>();
-        sshQualifierAttributes.put("id", qualifier);
-        return sshQualifierAttributes;
+    private void printLogContent(Writer writer, String qualifier, String sshLog) {
+        if (sshLog.length() > MAX_LENGTH) {
+            printLogFile(writer, sshLog, qualifier);
+        } else {
+            printBegin(writer, TEXT);
+            printCData(writer, sshLog);
+            printEnd(writer, TEXT);
+        }
     }
 
-    private Map<String, String> getSshTemplateAttributes(ReportContext reportContext, SshTemplate sshTemplate) {
-        Map<String, String> sshTemplateAttributes = new HashMap<>();
-        sshTemplateAttributes.put("startDate", reportContext.startExecutionZoned().toString());
-        sshTemplateAttributes.put("endDate", reportContext.endExecutionZoned().toString());
-        sshTemplateAttributes.put("host", sshTemplate.getSshSetting().getHostname() + ":" + sshTemplate.getSshSetting().getPort());
-        sshTemplateAttributes.put("logPath", sshTemplate.getSshSetting().getLogPath());
-        return sshTemplateAttributes;
+    private void printLogFile(final Writer writer, String logContent, String qualifier) {
+        File logFile = new File(new Date().getTime() + ".txt");
```

Comment:
I am personally not sure about the java.util.Date, but others might see it differently

Reply:
Well I see no reason why shouldn't use it. But it is us to you.